dev/src/graphics.cpp

```diff
@@ -179,6 +179,24 @@ nfr("gl_load_materials", "materialdefs,inline", "SI?", "S?",
         return err[0] ? Value(vm.NewString(err)) : NilVal();
     });
 
+nfr("gl_push_scissor", "x,y,width,height", "IIII", "",
+    "Enables scissor testing so only the pixels in the given rectangle can "
+    "be written.  (x,y) is the top left corner. Scissor state only lasts till "
+    "the end of the frame.",
+    [](StackPtr &, VM &vm, Value &x, Value &y, Value &w, Value &h) {
+   TestGL(vm);
+   PushScissorRect(int4(x.ival(), y.ival(), w.ival(), h.ival()));
+   return NilVal();
+});
+
+nfr("gl_pop_scissor", "", "", "",
```

Comment:
while this push & pop is the most "correct" design, it is not the simplest (needing a stack of values), and for such niche functionality, I prefer the simpler design of some of the other GL functions. I would suggest a single function that a) turns scissor OFF is size == screensize, otherwise on, and b) returns the previous existing scissor size if enabled (`glGet(GL_SCISSOR_BOX..)`), or returns the full screen size if disabled. That way, the code in `gl.lobster` can be the same as for `gl_color` etc.

Reply:
Fair enough.  I didn't end up nesting them anyway.  I've re-implemented in those terms, and cleaned up the style and indent problems.

Reply:
Also, I didn't change this because I didn't have a good idea for a test case, but it looks like the PopVec() calls for gl_start_text_input() are out of order.

Reply:
well, my design still allows nesting, just the "stack" for nesting will be taken care of by Lobster code instead of C++ code :)